lib/puppet/face/config.rb

```diff
@@ -119,6 +120,21 @@
         EOM
       end
 
+      # REMIND - we want to be able to make this check more specific to account
+      # for master node which is in fips mode then the fips restrictions
+      # should be based only when the setting is going to be agent specific
+      # and/or applicable to agent.
+      # It would depend on whether the digest_algorithm and supported_checksum_types
+      # settings are consumed by server.
+      if name == 'digest_algorithm'
+        Puppet::Settings.validate_digest_algorithm(value)
+      end
+
+      if name == 'supported_checksum_types'
+        values = value.split(/\s*,\s*/)
+        Puppet::Settings.validate_checksum_types(values)
+      end
+
```

Comment:
Puppet doesn't currently prevent invalid values from being set, so you can currently shoot yourself in the foot with:

```
$ bx puppet config set disable_i18n whoops
$ bx puppet apply -e "notify { 'hi': }"
Error: Could not initialize global default settings: Invalid value '"whoops"' for boolean parameter: disable_i18n
```

This one is a little different as `md5` is *sometimes* valid. That said, I think we should tackle the "is it a valid value" issue in a separate ticket.